### PR TITLE
Added Carousel Portal.

### LINKED ISSUE #4: Create carousel upload portal
Add it in the basic app and link to the admin panel. The user should be able to upload a list of images and also select the ones that will be actively shown at index.

Also add field for short description which may be utilised later.

### PR DESCRIPTION
**Pull request by Prakash2403**
fixes #4 
Note: I have used django-photologue library. Some bugs are there in library itself. So, before testing make sure that you have read `basic/README.md.`

### CODE REVIEW

**Comment by singhpratyush** [comment]
@Prakash2403 @yashLadha @Monal5031: Can you please set up this branch locally and test?

**Comment by Prakash2403** [comment]
I have tested it and it is working fine. Shall I post screenshots, @singhpratyush Sir?

**Comment by singhpratyush** [comment]
@Prakash2403: That would be helpful.

**Comment by Prakash2403** [comment]
**Index page**
 
<a href="http://www.imageupload.co.uk/image/BOL0"></a>

**Administrator index page**

<a href="http://www.imageupload.co.uk/image/BOLR"></a>

**List of all photos. `IS PUBLIC` field tells about the images which are visible on index page.**

<a href="http://www.imageupload.co.uk/image/BOLe"></a>

**Image upload Page.**

<a href="http://www.imageupload.co.uk/image/BOLl"></a>

File: wsgi/odyssy/basic/admin.py [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
 
-# Register your models here.
+from photologue.admin import PhotoAdmin as PhotoAdminDefault
+from photologue.models import Photo
+
+
+class PhotoAdmin(PhotoAdminDefault):
+    actions = ['set_active_image', 'remove_active_image']
+
+    def set_active_image(self, request, queryset):
```
Comment (singhpratyush) [inline wsgi/odyssy/basic/admin.py]: Is this an overridden method? In any case, use a proper annotation.

File: wsgi/odyssy/basic/admin.py [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+        queryset.update(is_public=True)
+
+    def remove_active_image(self, request, queryset):
```
Comment (singhpratyush) [inline wsgi/odyssy/basic/admin.py]: Same here.

File: wsgi/odyssy/basic/admin.py [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+        queryset.update(is_public=False)
+
+    set_active_image.short_description = "Show them on index page"
+    remove_active_image.short_description = "Remove them from index page"
+
+admin.site.unregister(Photo)
+admin.site.register(Photo, PhotoAdmin)
```
Comment (singhpratyush) [inline wsgi/odyssy/basic/admin.py]: Please resolve.

File: wsgi/odyssy/basic/templates/basic/index.html [hunk 4]
```diff
@@ -23,4 +23,9 @@
   {% for event in events %}
   <li><a href="{% url 'event-view-single' event.id %}">{{ event }}</a></li>
   {% endfor %}
+
+  Active Carousel images
+  {% for image in carousel %}
+  <li><img src=" media/{{image.image}}" alt="My image"/></li>
```
Comment (singhpratyush) [inline wsgi/odyssy/basic/templates/basic/index.html]: Add proper spacing with `{{` and `}}`.

File: wsgi/odyssy/odyssy/settings.py [hunk 5]
```diff
@@ -73,6 +79,10 @@
 
 ROOT_URLCONF = 'odyssy.urls'
 
+STATICFILES_DIRS = [
+    WSGI_DIR+"/media",
```
Comment (singhpratyush) [inline wsgi/odyssy/odyssy/settings.py]: Space around binary operator.

**Comment by Prakash2403** [comment]
> Is this an overridden method? In any case, use a proper annotation.

This is just a regular function. Example given in official documentation doesn't includes any annotations. That's why I haven't included any annotations here.

Please refer to official documentation
[Writing action functions](https://docs.djangoproject.com/en/1.11/ref/contrib/admin/actions/#writing-action-functions)
[Advanced Action Techniques](https://docs.djangoproject.com/en/1.11/ref/contrib/admin/actions/#advanced-action-techniques)

File: wsgi/odyssy/basic/admin.py (hunk 3 above)
Comment (Prakash2403) [inline wsgi/odyssy/basic/admin.py]: Any idea why this is poping up every time. I have added newline at the end.

Comment (singhpratyush) [inline wsgi/odyssy/basic/admin.py]: What type of EOL are you using? Make sure it is `Unix` type.

**Comment by singhpratyush** [comment]
In that case, since the `methods` do not rely on the state of the object, they can be `functions`. Removing `self` and adding `@staticmethod` would be more desirable.

File: wsgi/odyssy/basic/admin.py (hunk 3 above)
Comment (Prakash2403) [inline wsgi/odyssy/basic/admin.py]: I have edited every file using pycharm. I don't know why it is showing error only in this file.

**Comment by Prakash2403** [comment]
> Removing self and adding @staticmethod would be more desirable.

@singhpratyush sir, If I add `@staticmethod` annotation, then these two lines are not working. I can post the traceback if you want.

> set_active_image.short_description = "Show them on index page"
    remove_active_image.short_description = "Remove them from index page"

**Comment by singhpratyush** [comment]
@Prakash2403: Please post the traceback.

**Comment by Prakash2403** [comment]
``` 
Environment:


Request Method: POST
Request URL: http://127.0.0.1:8000/adminphotologue/photo/

Django Version: 1.11.1
Python Version: 2.7.9
Installed Applications:
('photologue',
 'sortedm2m',
 'events.apps.EventConfig',
 'news.apps.NewsConfig',
 'basic.apps.BasicConfig',
 'announcement.apps.AnnouncementConfig',
 'django.contrib.sites',
 'django.contrib.sitemaps',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')



Traceback:

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/contrib/admin/options.py" in wrapper
  551.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/contrib/admin/sites.py" in inner
  224.             return view(request, *args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/usr/local/lib/python2.7/dist-packages/django/contrib/admin/options.py" in changelist_view
  1568.                 response = self.response_action(request, queryset=cl.get_queryset(request))

File "/usr/local/lib/python2.7/dist-packages/django/contrib/admin/options.py" in response_action
  1286.             response = func(self, request, queryset)

Exception Type: TypeError at /adminphotologue/photo/
Exception Value: set_active_image() takes exactly 2 arguments (3 given)
```
**new admin.py**
```
from django.contrib import admin

from photologue.admin import PhotoAdmin as PhotoAdminDefault
from photologue.models import Photo


class PhotoAdmin(PhotoAdminDefault):
    actions = ['set_active_image', 'remove_active_image']

    @staticmethod
    def set_active_image(request, queryset):
        queryset.update(is_public=True)

    @staticmethod
    def remove_active_image(reuqest, queryset):
        queryset.update(is_public=False)

PhotoAdmin.set_active_image.short_description = "Show them on index page"
PhotoAdmin.remove_active_image.short_description = "Remove them from index page"
admin.site.unregister(Photo)
admin.site.register(Photo, PhotoAdmin)

```

**Comment by singhpratyush** [comment]
It looks like that those are methods of some superclass. If so, try annotating with `@override`.

**Pull request merged**